jcapi-commands.go

```diff
@@ -0,0 +1,158 @@
+	User             string   `json:"user,omitempty"`           // user to run as (000000000000000000000000 for root)
+	Files            []string `json:"files,omitempty"`          // list of files uploaded by the command
+	Systems          []string `json:"systems,omitempty"`        // systems to run the command on
+	Tags             []string `json:"tags,omitempty"`           // tags to run the command on (tags and systems are mutually exclusive)
+	LaunchType       string   `json:"launchType"`               // manual/add-delete-user/repeated/scheduled
+	ListensTo        string   `json:"listensTo"`                // AddUser/DeleteUser (when launchType is add-delete-user)
+	Schedule         string   `json:"schedule,omitempty"`       // immediate/agentEvent (launchType=add-delete-user)/a crontab(5) time entry as in "0 0 2 * * 6"
+	ScheduledRunDate string   `json:"scheduledRunDate"`         // when LaunchType='scheduled', set to the date on which to start the command
+	ScheduledRunTime string   `json:"scheduledRunTime"`         // when LaunchType='scheduled', set to the time at which to start the command
+	Trigger          string   `json:"trigger,omitempty"`        // generate trigger (No longer supported)
+	Timeout          string   `json:"timeout"`                  // Command time out in seconds, after which it will be killed
+	Organization     string   `json:"organization,omitempty"`   // organization ID for this command (auto-populated)
+	Sudo             bool     `json:"sudo"`                     // Indicates whether the command should be run with sudo
+
+	Skip  int `json:"skip"`  // Objects to skip on /search POST
+	Limit int `json:"limit"` // Max objects to return on /search POST
+}
+
+func (e JCCommand) ToString() string {
+	return fmt.Sprintf("command: %v", e)
+}
+
+func getJCCommandsFromResults(result []byte) (commands []JCCommand, err JCError) {
+	commandResults := JCCommandResults{}
+
+	err = json.Unmarshal(result, &commandResults)
+	if err != nil {
+		err = fmt.Errorf("Could not unmarshal result '%s', err='%s'", string(result), err.Error())
+	}
+
+	commands = commandResults.Results
+
+	return
+}
+
+func (jc JCAPI) GetAllCommands() (commandList []JCCommand, err JCError) {
+	var empty []byte
+
+	for skip := 0; skip == 0 || len(commandList) == searchLimit; skip += searchSkipInterval {
+		url := fmt.Sprintf("%s?sort=hostname&skip=%d&limit=%d", COMMAND_PATH, skip, searchLimit)
+
+		jcSysRec, err2 := jc.DoBytes(MapJCOpToHTTP(Read), url, empty)
+
+		if err2 != nil {
+			return nil, fmt.Errorf("ERROR: Get commands to JumpCloud failed, err='%s'", err2)
+		}
+
+		if jcSysRec == nil {
+			return nil, fmt.Errorf("ERROR: No commands found")
+		}
+
+		resultsBlock, err2 := getJCCommandsFromResults(jcSysRec)
+		if err2 != nil {
+			err = fmt.Errorf("Could not get resultsBlock data, err='%s'", err2.Error())
+			return
+		}
+
+		for i, _ := range resultsBlock {
+			if resultsBlock[i].Id != "" {
+				commandList = append(commandList, resultsBlock[i])
+			}
```

Comment:
Did you want to print a warning if the command has no Id?  Or is that an expected case?